examples/tutorials/core_api_pytorch_mnist/model_def_adaptive.py

```diff
@@ -211,18 +225,19 @@ def main(core_context):
     scheduler = StepLR(optimizer, step_size=1, gamma=args.gamma)
 
     # NEW: Iterate through the core_context.searcher.operations() to decide how long to train for.
-    starting_epoch = 0
-    epoch = starting_epoch
+    # Start with the number of epochs completed, in case of pausing and resuming the experiment.
+    epoch_idx = epochs_completed
     last_checkpoint_batch = None
 
     for op in core_context.searcher.operations():
 
         # NEW: Use a while loop for easier accounting of absolute lengths.
-        while epoch < op.length:
-            steps_completed = epoch * len(train_loader)
+        while epoch_idx < op.length:
 
             # NEW: Pass op into train() and test().
-            train(args, model, device, train_loader, optimizer, core_context, epoch, op)
+            train(args, model, device, train_loader, optimizer, core_context, epoch_idx, op)
```

Comment:
It's confusing in an example to pass `epoch_idx` into `train()`, but inside `train()` it's called `epoch` instead.  I really think you should update all instances of `epoch` to reflect the exact meaning (`epoch_idx` or `epochs_completed`), so quantities are easier to track across function calls.